### DISCUSSION TITLE
why my last_query option not work when i'm not give q(query parameter)?

### DISCUSSION THREAD

**Discussion by Ar-Anik** [fastapi/fastapi#6070] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Cookie, Depends, FastAPI
import uvicorn

app = FastAPI()

def query_extractor(q: str = None):
    return q

def query_or_cookie_extractor(q: str=Depends(query_extractor), last_query: str=Cookie(default=None)):
    print("Q: ", q)
    print("last_query: ", last_query)

    if not q:
        return last_query
    return q

@app.get("/items/")
async def read_query(query: str = Depends(query_or_cookie_extractor)):
    return {
        "query": query
    }

if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", port=int(os.getenv("9000")))
```

### Description

In fast api documentation say if query parameter q is null and give last_query this code show last_query parameter but this code show null when i am not give q query parameter why?

### Operating System

Linux

### FastAPI Version

0.70.1

### Python Version

3.8

**Comment by jgould22** [comment]
Are you submitting the Cookie correctly ?

The cookie default is None so if the cookie is not correct the code will still return None.

**Comment by Ar-Anik** [comment]
yes i submit from localhost:9000/docs

**Comment by jgould22** [comment, marked as answer]
Looks like this is a limitation of the swagger ui - https://github.com/swagger-api/swagger-js/issues/1163

If you try with the curl example it works but you are correct it doesnt work in the UI.
